bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/Wavelength.scala

```diff
@@ -0,0 +1,66 @@
+package edu.gemini.spModel.core
+
+/** Representation of wavelengths. */
+sealed trait Wavelength extends Serializable {
+
+  /** The wavelength as nanometers.
+    * @group Conversions
+    */
+  def toNanometers: Double
+
+  /** The wavelength as microns or micrometers.
+    * @group Conversions
+    */
+  def toMicrons: Double = toNanometers * 1000
+
+  /**
+   * Addition.
+   * @group Operations
+   */
+  def +(a: Wavelength): Wavelength =
+    Wavelength.fromNanometers(toNanometers + a.toNanometers)
+
+  /**
+   * Subtraction.
+   * @group Operations
+   */
+  def -(a: Wavelength): Wavelength =
+    Wavelength.fromNanometers(toNanometers - a.toNanometers)
+
+  /**
+   * Scalar multiplication.
+   * @group Operations
+   */
+  def *(factor: Double): Wavelength =
+    Wavelength.fromNanometers(toNanometers * factor)
+
+  /** @group Overrides */
+  final override def toString =
+    s"Wavelength(${toNanometers}nm)"
+
+  /** @group Overrides */
+  final override def equals(a: Any) =
+    a match {
+      case a: Wavelength => a.toNanometers == this.toNanometers
+      case _             => false
+    }
+
+  /** @group Overrides */
+  final override def hashCode =
+    toNanometers.hashCode
+}
+
+object Wavelength {
```

Comment:
I guess you could consider adding Monoid and Order definitions in here even though there is no use right away